packages/next/server/router.ts

```diff
@@ -225,10 +225,10 @@ export default class Router {
       ...this.headers,
       ...this.redirects,
       ...this.rewrites.beforeFiles,
+      ...this.fsRoutes,
```

Comment:
Are we wanting to change middleware coming after public files? If this was not intended we may need to change this handling. We should also add a test to assert our desired behavior one way or the other so we don't accidentally change this.

Reply:
We want to change dynamic middleware coming after public files. I fixed to invoke routes in the following order:
- internal fs routes (`/_next/**`)
- static middleware
- fs routes (only public and static routes)
- dynamic middleware

It's intended:
- any middleware does not match with `/_next/**`
- static middleware can match with public and static routes but dynamic ones cannot.

Reply:
Looks good, this order definitely seems more desirable, left a comment on the runtime PR but after that's landed we should be good to land this to canary.